python/cuml/tests/test_metrics.py

```diff
@@ -901,6 +901,71 @@ def prep_dense_array(array, metric, col_major=0):
         return np.asfortranarray(array) if col_major else array
 
 
+@pytest.mark.parametrize("metric", ["nan_euclidean"])
+@pytest.mark.parametrize("matrix_size", [(5, 4), (1000, 3), (2, 10),
+                                         (500, 400), unit_param((1000, 100))])
+def test_nan_euclidean_distances(metric: str, matrix_size):
```

Comment:
It looks like much of this is duplicated from the `test_pairwise_distance` pytest and this file is pretty large already. We should be able to combine this w/ the existing test and add nans to the input when the appropriate distance is invoked.

Reply:
I have made the other changes that were suggested, however in the test cases part to add nan_euclidean metric to test_pairwise_distance there should be a lot of the existing code which needs to be changed, because pairwise_distaces accept numpy arrays also as an input however nan_euclidean function accepts only cupy arrays.

Reply:
@Sreekiran096 I missed that detail when reviewing this PR. We should be accepting `numpy` arrays as well, which should be as easy as putting `X = cp.asarray(X)` at the beginning (it's a no-op when the data is already on the GPU).